pkg/e2e/verify/user-workload-monitoring.go

```diff
@@ -0,0 +1,238 @@
+
+	"github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	kv1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/rest"
+
+	"github.com/openshift/osde2e/pkg/common/alert"
+	"github.com/openshift/osde2e/pkg/common/helper"
+	"github.com/openshift/osde2e/pkg/common/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var userWorkloadMonitoringTestName string = "[Suite: informing] [OSD] User Workload Monitoring"
+
+func init() {
+	alert.RegisterGinkgoAlert(userWorkloadMonitoringTestName, "SD-SREP", "Max Whittingham", "sd-cicd-alerts", "sd-cicd@redhat.com", 4)
+}
+
+var _ = ginkgo.Describe(userWorkloadMonitoringTestName, func() {
+	h := helper.New()
+	const (
+		prometheusName = "prometheus-example-app"
+	)
+
+	userName := util.RandomStr(5) + "@customdomain"
+	identities := []string{"otherIDP:testing_string"}
+	groups := []string{"dedicated-admins"}
+	uwmtestns := util.RandomStr(5)
+
+	ginkgo.Context("User Workload Monitoring", func() {
+		ginkgo.It("has the required prerequisites for testing", func() {
+			//Create a new user that will have dedicated-admin privileges, add the user to the dedicated-admins group
+			_, err := createUser(userName, identities, groups, h)
+			Expect(err).NotTo(HaveOccurred(), "Could not create user workload monitoring testing user")
+			//Add the user to the dedicated-admin group
+			_, err = addUserToGroup(userName, groups[0], h)
+			Expect(err).NotTo(HaveOccurred(), "Could not grant dedicated-admin permissions to user workload monitoring user")
+			//Create a namespace to run the tests in
+			_, err = createNamespace(uwmtestns, h)
+			Expect(err).NotTo(HaveOccurred(), "Could not create user workload monitoring testing namespace")
+			//Launch a pod & service as the targets of the ServiceMonitor and PrometheusRules objects
+			pod := newPod(prometheusName, uwmtestns, "quay.io/brancz/prometheus-example-app:v0.2.0")
+			err = createPodUwm(pod, uwmtestns, h)
+			Expect(err).NotTo(HaveOccurred(), "Could not create user workload monitoring testing pod")
+			_, err = GenerateService(8080, prometheusName, uwmtestns, prometheusName)
+			Expect(err).NotTo(HaveOccurred(), "Could not create user workload monitoring testing service")
+		})
+
+		ginkgo.It("has create access to the user-workload-monitoring-config configmap", func() {
+			h.Impersonate(rest.ImpersonationConfig{
+				UserName: userName,
+				Groups:   []string{"system:authenticated", "system:authenticated:oauth", "dedicated-admins"},
+				Extra:    map[string][]string{},
+			})
+			// Need a test to verify create/edit access to the user-workload-monitoring-config configmap
+			uwme2ecm := newUwmCm("user-workload-monitoring-config", "openshift-user-workload-monitoring")
+			edituwmcm, err := h.Kube().CoreV1().ConfigMaps("openshift-user-workload-monitoring").Create(context.TODO(), uwme2ecm, metav1.CreateOptions{})
```

Comment:
this seems not a big deal but upon the second run against a staging cluster, the suite fails because of _user-workload-monitoring-config_ configmap is already existed considering the previous run. I would delete this with a privileged user as to left nothing on the cluster upon successful run. However, it can also stay as it is if deletion is not our consideration

Reply:
Deletion should also be tested so probably good to add it anyway